### PR TITLE
Build order rules

### PR DESCRIPTION
**Pull request by SchrodingersGat**
- Adds new option preventing build orders from being completed if there are open child orders
- Also provides improvement to documentation of settings
- Settings definitions can now be called directly in the documentation

### CODE REVIEW

File: docs/docs/settings/global.md [hunk 1]
```diff
@@ -17,120 +17,124 @@ Global settings are arranged in the following categories:
...
-| --- | --- | --- | --- |
-| Enable registration | Boolean | Enable self-registration for users on the login-pages | False |
-| Enable SSO | Boolean | Enable SSO on the login-pages | False |
-| Enable SSO registration | Boolean | Enable self-registration for users via SSO on the login-pages | False |
-| Enable SSO group sync | Boolean | Enable synchronizing InvenTree groups directly from the IdP | False |
-| SSO group key | String | The name of the groups claim attribute provided by the IdP | |
-| SSO group map | String (JSON) | A mapping from SSO groups to local InvenTree groups | {} |
-| Remove groups outside of SSO | Boolean | Whether groups assigned to the user should be removed if they are not backend by the IdP. Disabling this setting might cause security issues | True |
-| Enable password forgot | Boolean | Enable password forgot function on the login-pages.<br><br>This will let users reset their passwords on their own. For this feature to work you need to configure E-mail | True |
-| E-Mail required | Boolean | Require user to supply e-mail on signup.<br><br>Without a way (e-mail) to contact the user notifications and security features might not work! | False |
```
Comment (matmair) [inline docs/docs/settings/global.md:45]: This description is mostly lost

File: docs/docs/settings/global.md [hunk 2]
```diff
@@ -17,120 +17,124 @@ Global settings are arranged in the following categories:
...
-| Enforce MFA | Boolean | Users must use multifactor security.<br><br>This forces each user to setup MFA and use it on each authentication | False |
-| Mail twice | Boolean | On signup ask users twice for their mail | False |
-| Password twice | Boolean | On signup ask users twice for their password | True |
-| Auto-fill SSO users | Boolean | Automatically fill out user-details from SSO account-data.<br><br>If this feature is enabled the user is only asked for their username, first- and surname if those values can not be gathered from their SSO profile. This might lead to unwanted usernames bleeding over. | True |
```
Comment (matmair) [inline docs/docs/settings/global.md:49]: This description is mostly lost

File: docs/docs/settings/user.md [hunk 3]
```diff
@@ -32,24 +32,44 @@ This screen allows the user to customize display of items on the InvenTree home
...
+{{ usersetting("SEARCH_PREVIEW_EXCLUDE_INACTIVE_RETURN_ORDERS") }}
 
 ### Notifications
 
-Settings related to notification messages
+Settings related to notification messages:
 
-{% with id="user-notification", url="settings/user_notifications.png", description="User Notification Settings" %}
-{% include 'img.html' %}
-{% endwith %}
```
Comment (matmair) [inline docs/docs/settings/user.md:47]: This picture is still in the repo but not referenced anywhere

File: docs/docs/settings/user.md [hunk 4]
```diff
@@ -32,24 +32,44 @@ This screen allows the user to customize display of items on the InvenTree home
...
+{{ usersetting("NOTIFICATION_ERROR_REPORT") }}
 
 ### Reporting
 
-Settings for label printing and report generation
+Settings for label printing and report generation:
 
-{% with id="user-reporting", url="settings/user_reporting.png", description="User Reporting Settings" %}
-{% include 'img.html' %}
-{% endwith %}
```
Comment (matmair) [inline docs/docs/settings/user.md:55]: This picture is still in the repo but not referenced anywhere

File: docs/main.py [hunk 5]
```diff
@@ -219,3 +235,45 @@ def templatefile(filename):
         )
 
         return includefile(fn, f'Template: {base}', format='html')
+
+    @env.macro
+    def settingtableheader():
```
Comment (matmair) [inline docs/main.py:240]: This seems to not be used in the PR elsewhere

File: tasks.py [hunk 6]
```diff
@@ -1404,8 +1414,16 @@ def check_already_current(tag=None, sha=None):
         'compile_schema': 'Compile the schema documentation first (default: False)',
     }
 )
-def docs_server(c, address='localhost:8080', compile_schema=False):
+def docs_server(
+    c, address='localhost:8080', extract_settings=True, compile_schema=False
+):
     """Start a local mkdocs server to view the documentation."""
+    if extract_settings:
```
Comment (matmair) [inline tasks.py:1421]: Q (nb): Do we need this option? Is there a case where the settings file is not required for rendering?

**Review by matmair — COMMENTED** [review]
Intresting idea with the settings, could improve the workflow. Af places lose a bit of information - maybe we can address those.

Sidenote: The tasks get more and more crowded. Mabe we should spilt them into 3 [namespaces](https://docs.pyinvoke.org/en/stable/concepts/namespaces.html)? The root one for things users might want to interact with, 1 for development stuff and 1 for internal things needed for builds etc.

**Comment by SchrodingersGat** [comment]
@matmair namespaces is an interesting idea for the invoke tasks, there are certainly a lot now! Do you want to make an issue to address thsi?

**Pull request merged**